src/lexer/lex.c

```diff
@@ -218,6 +224,46 @@ int real_lex(Lexer *l, Token *t) {
     }
 
     // TODO - parse character or string literal
+    int c;
+    int *type;
```

Comment:
I think it might be clearer if we used named constants here instead of integer literals for the values of `type`. Also, is there a reason for the usage of an `int*`?